lib/commands/actions.js

```diff
@@ -250,11 +250,19 @@ helpers.getScreenshotDataWithAdbShell = async function (adb, opts) {
 };
 
 helpers.getScreenshotDataWithAdbExecOut = async function (adb) {
-  let {stdout} = await exec(adb.executable.path, ['exec-out', '/system/bin/screencap -p'],
-                            {encoding: 'binary', isBuffer: true});
+  let {stdout, stderr, code} = await exec(adb.executable.path,
+                                    adb.executable.defaultArgs
+                                      .concat(['exec-out', '/system/bin/screencap', '-p']),
+                                    {encoding: 'binary', isBuffer: true});
+  // if there is an error, throw
+  if (code || stderr.length) {
```

Comment:
I assume exec throws an error automatically if code !== 0

Reply:
It should.

Reply:
which means the interpreter will never reach this block of code if code !== 0

Reply:
Perhaps.